prov/util/src/util_cq.c

```diff
@@ -261,6 +261,37 @@ static const char *util_cq_strerror(struct fid_cq *cq, int prov_errno,
 	return fi_strerror(prov_errno);
 }
 
+int ofix_cq_read(const struct fi_provider *prov, struct util_cq *util_cq,
+		struct fid_cq *cq, struct fi_cq_tagged_entry *comp)
+{
+	struct util_cq_err_entry *entry;
+	int ret;
+
+	ret = fi_cq_read(cq, comp, 1);
+	if (ret == -FI_EAVAIL) {
+		entry = calloc(1, sizeof(*entry));
+		if (!entry) {
+			FI_WARN(prov, FI_LOG_CQ,
+					"Unable to allocate util_cq_err_entry\n");
+			return -FI_ENOMEM;
+		}
+		ret = fi_cq_readerr(cq, &entry->err_entry, 0);
+		if (ret < 0) {
```

Comment:
It looks like any error entry read out from the msg-cq (underlying CQ) is added to the util-cq. 
Just wondering if this can be applied for all operations. What if the error-entry was for an internal control message?